lib/eclipse/include/opm/parser/eclipse/EclipseState/Aquifer.hpp

```diff
@@ -0,0 +1,170 @@
+#include <opm/parser/eclipse/Deck/Deck.hpp>
+#include <opm/parser/eclipse/Deck/DeckRecord.hpp>
+#include <opm/parser/eclipse/Deck/DeckKeyword.hpp>
+
+#include <opm/parser/eclipse/EclipseState/Tables/Aqudims.hpp>
+#include <opm/parser/eclipse/EclipseState/Tables/TableContainer.hpp>
+#include <opm/parser/eclipse/EclipseState/Tables/AqutabTable.hpp>
+#include <boost/concept_check.hpp>
+
+namespace Opm {
+  class Deck;
+    class Aquifer {
+    public:
+
+        Aquifer() :
+            m_aquId(0),
+            m_datDepth(0.0),
+            m_aqPerm(0.0),
+            m_aqPoro(0.0),
+            m_aqCt(0.0),
+            m_aqRad(0.0),
+            m_aqThickness(0.0),
+            m_aqInfluenceAngle(0.0),
+            m_waterPressTableNum(0 ),
+            m_influenceFnTableNum( 0 )
+        {
+	  Deck init_deck = Deck();
+	  deck_ = &init_deck;
+	}
+
+        explicit Aquifer(const Deck& deck) :
+            Aquifer()
+	    {
+	      deck_ = deck;
+	      if (deck.hasKeyword("AQUCT")) {
+		  const auto& record = deck.getKeyword( "AQUCT" , 0 ).getRecord( 0 );
+		  m_aquId  = record.getItem("AQUIFER_ID").get<int>(0);
+		  m_datDepth  = record.getItem("DAT_DEPTH").get<double>(0);
+		  m_aqPerm  = record.getItem("PERM_AQ").get<double>(0);
+		  m_aqPoro  = record.getItem("PORO_AQ").get<double>(0);
+		  m_aqCt  = record.getItem("C_T").get<double>(0);
+		  m_aqRad  = record.getItem("RAD").get<double>(0);
+		  m_aqThickness  = record.getItem("THICKNESS_AQ").get<double>(0);
+		  m_aqInfluenceAngle  = record.getItem("INFLUENCE_ANGLE").get<double>(0);
+		  m_waterPressTableNum  = record.getItem("TABLE_NUM_WATER_PRESS").get<int>(0);
+		  m_influenceFnTableNum  = record.getItem("TABLE_NUM_INFLUENCE_FN").get<int>(0);
+	      }
+	      
+	      int number_tables = m_aqudims.NIFTBL;
+	      
+	      
+            
+
+        
+        }
+
+        size_t getAquId() const
+        {
+            return m_aquId;
+        }
+
+        size_t getAqDatumDepth() const
+        {
+            return m_datDepth;
+        }
+
+        size_t getAqPermeability() const
+        {
+            return m_aqPerm;
+        }
+
+        size_t getAqPorosity() const
+        {
+            return m_aqPoro;
+        }
+
+        size_t getAqTotalCompressibility() const
+        {
+            return m_aqCt;
+        }
+
+        size_t getAqRadius() const
+        {
+            return m_aqRad;
+        }
+        
+        size_t getAqThickness() const
+        {
+            return m_aqThickness;
+        }
+
+        size_t getAqInfluenceAngle() const
+        {
+            return m_aqInfluenceAngle;
+        }
+
+        size_t getAqWaterPressureTableNumber() const
+        {
+            return m_waterPressTableNum;
+        }        
+
+        size_t getAqInfluenceFunctionTableNumber() const
+        {
+            return m_influenceFnTableNum;
+        }        
+      
+        std::vector<double> getTimeColumn() const{
+	if (deck_.hasKeyword("AQUTAB")) {
```

Comment:
This just shows that basing the implementation on the `TableManager` is wrong. My suggestion is that the `Aquifer` class is based on something like:
```C++
typedef struct {
     std::vector<double> time;
     std::vector<double> p;
} influence_fn;

std::vector<influence_fn> functions;
```

This can of course be implemented in many ways, and personally I think it would have been nice to return a callable instance instead of raw `std::vector<double>` - but anyway the current implementation which is based on storing a private `Deck&` and checking that to get logic right - will not be merged. Sorry.